### PR TITLE
Feature/improve documentation 5.x

### PR DESCRIPTION
**Pull request by AlexWattrelos**
Please review especially the new configure OCR article, I am not 100% confident there.

### CODE REVIEW

File: docs/config-ocr.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Motivation
+
+OCR enables text recognition for your mobile scanning apps. This means you can add text recognition to your organization’s mobile data capture workflows. As a result, you can seamlessly switch between reading barcodes and text with the touch of a button. An OCR sample is shipped in the zip file of our OCR SDK.
+
+## License Key
+
+A dedicated OCR License key and SDK should be used. Please contact us (https://support.scandit.com/hc/en-us/requests/new) for more details.
+
+## Implementing OCR
+To integrate the OCR  into your Cordova app, follow the simple steps below:
```
Comment (TomekJurkowski) [inline docs/config-ocr.md]: double space after OCR ;)

File: docs/config-ocr.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+#### Add callbacks to handle the scanning event
+
+You now need to define the functions that are referenced in the show() call. All functions take one argument, the manual
```
Comment (TomekJurkowski) [inline docs/config-ocr.md]: the manual ? aren't we missing the rest of the sentence here?

File: docs/config-ocr.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		alert("Failed: " + error);
+	}
+
+~~~~~~~~~~~~~~~~
+
+#### Set the text scan settings
+
+- Create a Scan Settings and Text Recognition Settings
+- Set the recognition mode to “Text”
+- Specify the format/structure of the text to be scanned: regular expression which matches your text and white list of recognizable characters.
```
Comment (TomekJurkowski) [inline docs/config-ocr.md]: please use the dot at the end of a sentence in all four of these points, or in none of them :)

File: docs/config-ocr.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- Set the area in which text is to be recognized
+
+~~~~~~~~~~~~~~~~{.java}
+let settings = new Scandit.ScanSettings();
+settings.recognitionMode = Scandit.ScanSettings.RecognitionMode.TEXT;
+settings.textRecognition = new Scandit.TextRecognitionSettings();
+
+settings.textRecognition.regex = '([A-Z]{2}[0-9]{2}\\s([0-9]{4}\\s){4}([A-Z0-9]{1}))';
```
Comment (TomekJurkowski) [inline docs/config-ocr.md]: the whitelisting part seems to be missing in the sample

File: docs/examples.md [hunk 5]
```diff
@@ -1,6 +1,13 @@
-Example code     {#cordova-examples}
+Run the samples {#cordova-examples}
 ===================================
 
+The Scandit Barcode Scanner SDK (downloadable from https://ssl.scandit.com/sdk) comes with three demos:
+* Simple: shows how to use the barcode scanner in a cordova project
```
Comment (TomekJurkowski) [inline docs/examples.md]: dot

File: docs/examples.md [hunk 6]
```diff
@@ -9,113 +16,82 @@ In order to build the sample apps you must import one of them upon creation of y
...
+* OCR Sample: <path-to-repository>/samples/ocr/src/providers/scanner.ts
+* Ionic Sample: <path-to-repository>/samples/ionic-sample/src/providers/scanner-service/scanner.ts
 
-For the extended sample, the license key is in `<path-to-repository>/samples/extended/src/providers/scanner.ts`, in the `setAppKey` method. After setting the license key, running `npm run build` is necessary for the build files in the `www` folder to be updated — the recommendation is to add the license key before running the commands below.
 
+### Build the sample
+
+#### Build the Simple or MatrixScan Sample
+
+Go to the folder directory that should contain your project.
```
Comment (TomekJurkowski) [inline docs/examples.md]: `Go to the folder` or  `Go to the directory` but not  `Go to the folder directory`

File: docs/integrate.md [hunk 7]
```diff
@@ -29,11 +29,11 @@ Use the cordova CLI to add the plugin to your already existing project.
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.java}
     cd <directory of your project>
-	cordova plugin add <path to downloaded and unzipped plugin>
+	  cordova plugin add <path to downloaded and unzipped plugin>
```
Comment (TomekJurkowski) [inline docs/integrate.md]: can you replace the tab with spaces? the indentation should be the same as for the previous line

File: docs/integrate.md [hunk 8]
```diff
@@ -147,6 +145,18 @@ Compile your project. Attach a device and run the app on your desired plattform.
     cordova run windows
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+## Build with PhoneGap Build (cloud service)
+
+Our plugin is not part of the plugins that Phonegap Build supports as it only supports open source plugins. If you want to use PhoneGap Build, you will have to add our plugin yourself.
```
Comment (TomekJurkowski) [inline docs/integrate.md]: `Our plugin is not part of the plugins that Phonegap Build supports as it only supports open source plugins.` I think I would rephrase that to sth simpler, e.g.:
`Our plugin is not supported by Phonegap Build, as it only supports open source plugins.`
Thoughts?

File: docs/integrate.md [hunk 9]
```diff
@@ -147,6 +145,18 @@ Compile your project. Attach a device and run the app on your desired plattform.
...
+
+Steps to use our plugin with Phonegap Build:
+1. Download the Barcode Scanner SDK for Phonegap (https://ssl.scandit.com/sdk)
```
Comment (TomekJurkowski) [inline docs/integrate.md]: dot (also in point 5.)

File: docs/scan-case.md [hunk 10]
```diff
@@ -1,14 +1,14 @@
-Using the Scandit Scan Case in your app      {#cordova-scan-case}
+Use the Scandit Scan Case in your app      {#cordova-scan-case}
 =========================================
- 
-This guide show how to scan barcodes with the Scandit scan case using the ScanCase API. 
 
-Using the ScanCase API is relatively easy: all you need to do is to create an ScanCase object and implement 
+This guide show how to scan barcodes with the Scandit scan case using the ScanCase API.
+
+Using the ScanCase API is relatively easy: all you need to do is to create an ScanCase object and implement
```
Comment (TomekJurkowski) [inline docs/scan-case.md]: typo: `an ScanCase` ---> `a ScanCase`

File: docs/integrate.md (hunk 8 above)
Comment (TomekJurkowski) [reply docs/integrate.md]: or maybe sth different, I just don't like the `part of the plugins`

File: docs/config-ocr.md (hunk 1 above)
Comment (AlexWattrelos) [reply docs/config-ocr.md]: how the hell do you see that? :D Done

File: docs/config-ocr.md (hunk 2 above)
Comment (AlexWattrelos) [reply docs/config-ocr.md]: The current docs was incomplete and I did a stupid copy paste. Corrected this on both articles.

File: docs/config-ocr.md (hunk 3 above)
Comment (AlexWattrelos) [reply docs/config-ocr.md]: Done

File: docs/examples.md (hunk 5 above)
Comment (AlexWattrelos) [reply docs/examples.md]: Done

File: docs/examples.md (hunk 6 above)
Comment (AlexWattrelos) [reply docs/examples.md]: Indeed :D Done

File: docs/config-ocr.md (hunk 4 above)
Comment (AlexWattrelos) [reply docs/config-ocr.md]: Done

File: docs/integrate.md (hunk 7 above)
Comment (AlexWattrelos) [reply docs/integrate.md]: Done

File: docs/integrate.md (hunk 8 above)
Comment (AlexWattrelos) [reply docs/integrate.md]: I do not like to say explicitly that it does not support us (as it is not really true). I changed it to 
`Phonegap Build only supports open source plugins. If you want to use PhoneGap Build, you will have to add our plugin yourself.`

File: docs/integrate.md (hunk 9 above)
Comment (AlexWattrelos) [reply docs/integrate.md]: Done

File: docs/scan-case.md (hunk 10 above)
Comment (AlexWattrelos) [reply docs/scan-case.md]: Done

Comment (AlexWattrelos) [reply docs/scan-case.md]: @TomekJurkowski, all done, commit was pushed

Comment (AlexWattrelos) [reply docs/scan-case.md]: Thanks for the careful review :)

File: docs/config-ocr.md (hunk 2 above)
Comment (TomekJurkowski) [reply docs/config-ocr.md]: awesome ;)

File: docs/config-ocr.md (hunk 1 above)
Comment (TomekJurkowski) [reply docs/config-ocr.md]: :D it's a gift (and a burden) I was born with

File: docs/integrate.md (hunk 8 above)
Comment (TomekJurkowski) [reply docs/integrate.md]: perfect, this sounds much better
